src/compiler/commandLineParser.ts

```diff
@@ -884,13 +888,306 @@ namespace ts {
+                    type: "object",
+                    elementOptions: commandLineOptionsToMap(typeAcquisitionDeclarations),
+                    extraKeyDiagnosticMessage: Diagnostics.Unknown_type_acquisition_option_0
+                },
+                {
+                    name: "extends",
+                    type: "string"
+                },
+                {
+                    name: "files",
+                    type: "list",
+                    element: {
+                        name: "files",
+                        type: "string"
+                    }
+                },
+                {
+                    name: "include",
+                    type: "list",
+                    element: {
+                        name: "include",
+                        type: "string"
+                    }
+                },
+                {
+                    name: "exclude",
+                    type: "list",
+                    element: {
+                        name: "exclude",
+                        type: "string"
+                    }
+                },
+                compileOnSaveCommandLineOption
+            ]);
+        }
+        return _tsconfigRootOptions;
+    }
+
+    interface JsonConversionNotifier {
+        /**
+         * Notifies parent option object is being set with the optionKey and a valid optionValue
+         * Currently it notifies only if there is element with type object (parentOption) and
+         * has element's option declarations map associated with it
+         * @param parentOption parent option name in which the option and value are being set
+         * @param option option declaration which is being set with the value
+         * @param value value of the option
+         */
+        onSetValidOptionKeyValueInParent(parentOption: string, option: CommandLineOption, value: CompilerOptionsValue): void;
+        /**
+         * Notify when valid root key value option is being set
+         * @param key option key
+         * @param keyNode node corresponding to node in the source file
+         * @param value computed value of the key
+         * @param ValueNode node corresponding to value in the source file
+         */
+        onSetValidOptionKeyValueInRoot(key: string, keyNode: PropertyName, value: CompilerOptionsValue, valueNode: Expression): void;
+        /**
+         * Notify when unknown root key value option is being set
+         * @param key option key
+         * @param keyNode node corresponding to node in the source file
+         * @param value computed value of the key
+         * @param ValueNode node corresponding to value in the source file
+         */
+        onSetUnknownOptionKeyValueInRoot(key: string, keyNode: PropertyName, value: CompilerOptionsValue, valueNode: Expression): void;
+    }
+
+    /**
+     * Convert the json syntax tree into the json value
+     */
+    export function convertToObject(sourceFile: JsonSourceFile, errors: Diagnostic[]): any {

```

Comment:
Should this be `@internal`?
Also, I'd recommend using `errors: Push<Diagnostic>`, since it's really an output, not an input.
It would be a good idea to separate parse tree -> object conversion and options validation. This function does both at once.